nexus_constructor/transformation.py

```diff
@@ -0,0 +1,63 @@
+import h5py
+from nexus_constructor.vector import Vector
+from abc import ABC, abstractmethod
+
+
+class Transformation(ABC):
+    """
+    Parent class for transformations on the model.
+    """
+
+    @abstractmethod
+    def __init__(self, name):
+        """
+        Creates an in-memory nexus file with a named entry containing an NXTransformation attribute.
+        :param name: The name of the root entry in the nexus file.
+        """
+        self.nexus_file = h5py.File(name, driver="core", backing_store=False)
```

Comment:
Rather than disabling tests, you could use a uuid here instead of `name` as the name of the file.
```python
import uuid
filename = str(uuid.uuid4())
```